app/controllers/spree/api/v1/omniauth_callbacks_controller.rb

```diff
@@ -0,0 +1,50 @@
+
+  class Spree::Api::V1::OmniauthCallbacksController < Devise::OmniauthCallbacksController
+    include Spree::Core::ControllerHelpers::Common
+    include Spree::Core::ControllerHelpers::Order
+    include Spree::Core::ControllerHelpers::Auth
+    include Spree::Core::ControllerHelpers::Store
+    skip_before_action :verify_authenticity_token
+
+    def login
+      eligible_providers = SpreeSocial::OAUTH_PROVIDERS.map { |p| p[1] if p[2] == 'true' }.compact
+
+      if !(eligible_providers.include?(auth_hash['provider']))
+        render json: {error: I18n.t('devise.omniauth_callbacks.provider_not_found', kind: auth_hash['provider'])}, status: :unprocessable_entity
```

Comment:
can we use failure_message method from https://github.com/heartcombo/devise/blob/master/app/controllers/devise/omniauth_callbacks_controller.rb ?

Reply:
we should be able to just raise error and base controller should be able to handle the exceptions..
we have lots of things going inside this action